src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -367,11 +368,24 @@ public override SyntaxNode AsPublicInterfaceImplementation(SyntaxNode declaratio
             });
         }
 
+        public interface Foo
+        {
+            void Moo<T>() where T : IEnumerable<T>;
+        }
+
+        public class Bar : Foo
+        {
+            public void Moo<T>() where T : IEnumerable<T>
+            {
+            }
+        }
+
```

Comment:
Apart from the name choice, they could live on in the tests, but we probably shouldn't have them here in the product code :smile: